File: test/solutions/190.jl
function reverseBits(n::UInt32)::UInt32
    result = 0
    for i in 0:31
        bit = (n >> i) & 1
        result = result | (bit << (31 - i))
    end
    return result
end


@testset "reverseBits tests" begin
    # Test case 1
    n = 0b00000010100101000001111010011100
    expected_output = 0b00111001011110000010100101000000
    @test reverseBits(n) == expected_output
    
    # Test case 2
    n = 0b11111111111111111111111111111101
    expected_output = 0b10111111111111111111111111111111
    @test reverseBits(n) == expected_output
end

File: test/rust/tests/190.rs
pub fn reverse_bits(n: u32) -> u32 {
    let mut result = 0;
    let mut n = n;
    for _ in 0..32 {
        result = result << 1;
        if n & 1 == 1 {
            result |= 1;
        }
        n = n >> 1;
    }
    result
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_example1() {
        assert_eq!(reverse_bits(0b00000010100101000001111010011100), 0b00111001011110000010100101000000);
    }

    #[test]
    fn test_example2() {
        assert_eq!(reverse_bits(0b11111111111111111111111111111101), 0b10111111111111111111111111111111);
    }
}
